src/bin/taker.rs

```diff
@@ -94,10 +80,30 @@ enum Commands {
         #[clap(name = "fee")]
         fee: u64,
     },
-    /// Sync the offer book
-    SyncOfferBook,
+
+    /// List all offers in taker's offerbook
+    ListOffers,
     /// Initiate the coinswap process
-    DoCoinswap,
+    DoCoinswap {
+        /// Sets the send amount.
+        #[clap(name = "send_amount")]
+        send_amount: u64,
+        /// Sets the fee-rate.
+        #[clap(name = "fee_rate")]
+        fee_rate: u64,
+        /// Sets the required on-chain confirmations.
+        #[clap(name = "required_confirms")] // TODO: Should we have default as 1?
+        required_confirms: u64,
+        /// Sets the maker count to initiate coinswap with.
+        #[clap(name = "maker_count", default_value = "2")]
+        // THINK: I do not want to  discard its default value as `maker_count =2`` resembles the coinswap protocol example.
+        // But due to which this arg has to kept after the required ones -> so will it makes sense to have this arg at the end even though
+        // It's the most important aspect of `do-coinswap`?
+        maker_count: usize,
+        /// Sets the transaction count.
+        #[clap(name = "tx_count", default_value = "3")]
+        tx_count: u32,
```

Comment:
Lets make default to 1. Also this need a good explainer doc. Setting it's value `n>1` will create "split-swaps". Which takes 1 or more utxos, and gives `n` new utxos back. 

Setting it to 1 simulates a "merge_swap`, which takes 1 or more utxos, and only give 1 new utxo back.  

Splits swaps are more costly than merge swaps. A 3tx swap will be about 3 times more costly than a 1tx swap. This will need to be improved later to reduce coinswap cost. Its unreasonably high right now.